assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3019,6 +3021,16 @@ protected String build_instructor_new_edit_assignment_context(VelocityPortlet po
                 }
                 context.put("groupsWithUserSubmission", groupsWithUserSubmission);
             }
+        }else if (isAssignmentsByRelatedGroups){
+            try{
+                Site site = siteService.getSite(siteId);
+                User user = (User) state.getAttribute(STATE_USER);
+                Collection<Group> relatedGroupsToUser = site.getGroupsWithMember(user.getId());
+
+                context.put("relatedGroupsToUser", relatedGroupsToUser);
+            }catch (IdUnusedException e) {
+                log.warn("Site not found for site id:{}", siteId);
+            }
```

Comment:
We need to rewrite all of this, let me make a commit to show it to you.

Reply:
Ok